library/Vanilla/Embeds/AbstractEmbed.php

```diff
@@ -0,0 +1,274 @@
+<?php
+/**
+ * @copyright 2009-2018 Vanilla Forums Inc.
+ * @license GPLv2
+ */
+
+namespace Vanilla\Embeds;
+
+use Exception;
+use Garden\Http\HttpRequest;
+use Garden\Http\HttpResponse;
+use Vanilla\InjectableInterface;
+
+/**
+ * Basic embed class.
+ */
+abstract class AbstractEmbed implements InjectableInterface {
```

Comment:
I was fully ready to defend the use of an "abstract" prefix, because it's easier to tell the class isn't directly instantiable, just by seeing the name. However, you won me over with being able to declare types as `Embed`, instead of `AbstractEmbed`. :godmode:

Reply:
https://dstecca.files.wordpress.com/2017/02/hurray-its-weekend-magical-me-pn21xf-clipart.jpg?w=840